frontend/src/components/data-collection/DuplicatesPosting.vue

```diff
@@ -174,6 +292,67 @@ export default {
         this.isLoading = false;
       });
     },
+    async fireFormValidate() {
+      await this.$nextTick();
+      await this.validateForm();
+    },
+
+    async validateForm() {
+      await this.$nextTick();
+      const isValid = await this.$refs.closeForm.validate();
+      this.validForm = isValid.valid;
+    },
+    getActiveCollection() {
+      this.closingCollectionType = formatCollectionTypeCode(this.collectionObject?.collectionTypeCode);
+      this.newCollection.collectionType = COLLECTION_TYPE_CODE_MAPPING[this.closingCollectionType]?.next;
+      this.calculateSnapshotDate();
+    },
+    calculateSnapshotDate(){
+      let newSnapshotDate = LocalDate.parse(this.collectionTypeCodesMap.get(this.newCollection.collectionType.toUpperCase()).snapshotDate);
+      this.newCollection.snapshotDate = findUpcomingDate(newSnapshotDate.month(), newSnapshotDate.dayOfMonth()).toString();
+      this.newCollection.collectionYear = this.newCollection.snapshotDate.slice(0, 4);
+
+      this.calculateSubmissionDate();
+    },
+    calculateSubmissionDate(){
+      let newSubmissionDate= LocalDate.parse(this.newCollection.snapshotDate, DateTimeFormatter.ofPattern(this.datePattern));
+      let weekLater = newSubmissionDate.plusWeeks(1);
+      this.newCollection.submissionDueDate = weekLater.with(TemporalAdjusters.nextOrSame(DayOfWeek.FRIDAY)).atStartOfDay().format(DateTimeFormatter.ofPattern(this.datePattern));
+
+      this.calculateAdminDates();
+    },
+    calculateAdminDates(){
+      let baseDate = LocalDate.parse(this.newCollection.submissionDueDate, DateTimeFormatter.ofPattern(this.datePattern));
+      let twoWeeksLater = baseDate.plusWeeks(2);
+      this.newCollection.duplicationResolutionDueDate = twoWeeksLater.atStartOfDay().format(DateTimeFormatter.ofPattern(this.datePattern));
+
+      let threeWeeksLater = baseDate.plusWeeks(3);
+      this.newCollection.signoffDueDate = threeWeeksLater.atStartOfDay().format(DateTimeFormatter.ofPattern(this.datePattern));
+
+      this.fireFormValidate();
+    },
+   async closeCollection() {
+      const confirmation = await this.$refs.confirmCloseCollection.open('Close Collection.', null, {color: '#fff', width: 580, closeIcon: false, subtitle: false, dark: false, resolveText: 'Confirm', rejectText: 'Cancel'});
+      if (!confirmation) {
+        return;
+      }
+      this.isLoading = true;
+      ApiService.apiAxios.post(`${Routes.sdc.BASE_URL}/collection/${this.collectionID}/close-collection`)
```

Comment:
Looks like it's not passing the body @SodhiA1

Reply:
updated, Thanks @mightycox